tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py

```diff
@@ -0,0 +1,101 @@
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.zookeeper import ZookeeperService
+
+
+class StreamsBrokerDownResilience(Test):
+    """
+    This test validates that Streams is resilient to a broker
+    being down longer than specified timeouts in configs
+    """
+
+    inputTopic = "streamsResilienceSource"
+    outputTopic = "streamsResilienceSink"
+    num_messages = 5
+
+    def __init__(self, test_context):
+        super(StreamsBrokerDownResilience, self).__init__(test_context=test_context)
+        self.zk = ZookeeperService(test_context, num_nodes=1)
+        self.kafka = KafkaService(test_context,
+                                  num_nodes=1,
+                                  zk=self.zk,
+                                  topics={
+                                      self.inputTopic: {'partitions': 1, 'replication-factor': 1},
+                                      self.outputTopic: {'partitions': 1, 'replication-factor': 1}
+                                  })
+
+    def get_consumer(self):
+        return VerifiableConsumer(self.test_context,
+                                  1,
+                                  self.kafka,
+                                  self.outputTopic,
+                                  "stream-broker-resilience-verify-consumer",
+                                  max_messages=self.num_messages)
+
+    def get_producer(self):
+        return VerifiableProducer(self.test_context,
+                                  1,
+                                  self.kafka,
+                                  self.inputTopic,
+                                  max_messages=self.num_messages,
+                                  acks=1)
+
+    def assert_produce_consume(self, test_state):
+        producer = self.get_producer()
+        producer.start()
+
+        wait_until(lambda: producer.num_acked > 0,
+                   timeout_sec=30,
+                   err_msg="At %s failed to send messages " % test_state)
+
+        consumer = self.get_consumer()
+        consumer.start()
+
+        wait_until(lambda: consumer.total_consumed() > 0,
+                   timeout_sec=60,
+                   err_msg="At %s streams did not process messages in 60 seconds " % test_state)
+
+    def setUp(self):
+        self.zk.start()
+
+    def test_streams_resilient_to_broker_down(self):
+        self.kafka.start()
+
+        processor = StreamsBrokerDownResilienceService(self.test_context, self.kafka, "none")
+        processor.start()
+
+        # until KIP-91 is merged we'll only send 5 messages to assert Kafka Streams is running before taking the broker down
+        # After KIP-91 is merged we'll continue to send messages the duration of the test
+        self.assert_produce_consume("before_broker_stop")
+
+        node = self.kafka.leader(self.inputTopic)
+
+        self.kafka.stop_node(node)
+
+        time.sleep(70)
```

Comment:
How this value determined? Is that relevant to the streams configs?

EDIT: I guess what I'd like to suggest, is to let this value not be a hand coded one but inferred from the streams config so that in the future if we only need to change on place if ever we want to update the timeout values.

Reply:
But this is Python code. How would you infer it from `StreamsConfig`? Print at console and parse it? Sounds like overkill to me.

Reply:
Actually I'm thinking the other way around: define the broker down time once in python code, and infer the corresponding streams retry and timeout values and set it when starting the instance as additional key-value config pairs

Reply:
ack